app/services/popular_tasks.rb

```diff
@@ -1,5 +1,6 @@
 class PopularTasks
   CACHE_EXPIRATION = 24.hours # Set the cache expiration time
+  BACKUP_CACHE_EXPIRATION = 7.days # Backup cache can have a longer expiration
```

Comment:
I think we need to think a bit more about how this would work.

If the bigquery data was unavailable for more than 7 days then what happens? 

I can think of other ways to do it - but this feels like a problem that must have been solved many times before. i.e. Only expire the cache if fresh data is available to fill it.

Reply:
I've caught up now... the cache will expire regardless of whether or not the API responds so I understand the need for a backup. And I like the idea of writing to the backup at the same time as you fetch the fresh data.